### PR TITLE
Feature/implement architecture for financial transactions

### PR DESCRIPTION
**Pull request by DawidStuzynski**

### CODE REVIEW

**Review by dawciobiel — APPROVED** [review]
Looks good to me,
**Maybe** you should add annotation as well, i.e: `@Validated` on `FinancialTransactionService`.

**Review by MackowiakPiotr — DISMISSED** [review]
Looks good to me

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/Wallet.java [hunk 1]
```diff
@@ -47,6 +47,10 @@ public class Wallet implements Serializable {
     @DateTimeFormat(pattern = "yyyy-mm-dd hh:mm:ss")
     private Instant creationDate;
 
+    @OneToMany(mappedBy = "wallet")
+    @JoinColumn(name = "financial_transaction_id")
+    private List<FinancialTransaction> financialTransactions;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/Wallet.java:52]: Please rename the name of `financialTransactions` -> `financialTransactionList`

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/FinancialTransactionDTO.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package pl.byczazagroda.trackexpensesappbackend.dto;
+
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+public record FinancialTransactionDTO(Long id, FinancialTransactionType transactionType, BigDecimal amount,
```
[the remark this answers is not in the document]
Comment (DawidStuzynski) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/FinancialTransactionDTO.java]: Changes applied

**Review by odzio33 — APPROVED** [review]
After modify some object name looks good.

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by mirekgab — APPROVED** [review]

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Pull request merged**